tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/AdoNet/SystemDataSqlClientTests.cs

```diff
@@ -27,20 +29,37 @@ public SystemDataSqlClientTests(ITestOutputHelper output)
             SetServiceVersion("1.0.0");
         }
 
-        public static IEnumerable<object[]> GetEnabledConfig()
-            => from packageVersionArray in PackageVersions.SystemDataSqlClient
-               from metadataSchemaVersion in new[] { "v0", "v1" }
-               from propagation in new[] { string.Empty, "100", "randomValue", "disabled", "service", "full" }
-               from injectStoredProc in new[] { "true", "false" }
-               select new[] { packageVersionArray[0], metadataSchemaVersion, propagation, injectStoredProc };
+        public static string[] GetPackageVersion()
+        {
+            return [.. PackageVersions.SystemDataSqlClient.Select(p => p[0] as string)];
+        }
+
+        public static string[] GetMetadataSchemaVersions()
+        {
+            return ["v0", "v1"];
+        }
+
+        public static string[] GetDbmPropagationModes()
+        {
+            return [string.Empty, "100", "randomValue", "disabled", "service", "full"];
+        }
+
+        public static string[] GetInjectStoredProcedure()
+        {
+            return ["true", "false"];
+        }
```

Comment:
If you change `string injectStoredProc` to be `bool` instead, and just call `ToString()` in the method, you can drop this and rely on the built-in support for combinatorial I think?